client/src/components/cart/shipping/index.tsx

```diff
@@ -1,7 +1,195 @@
+import Button from '~/components/common/Button';
+import Divider from '~/components/common/Divider';
+import { alert } from '~/utils/modal';
+import ShipItem from '../shippItem';
+import useUser from '~/lib/hooks/useUser';
+import ShippingModal from '../shippingModal';
+import { ButtonWrapper, ShipHeader } from './index.style';
+
+const mockData = [
+  {
+    idx: 5,
+    name: '황병현',
+    phone: '010-5022-2332',
+    code: '123213',
+    address: '서울 영등포구 선유로 200가길 39',
+    detailAddress: '우형빌딩200호',
+    selected: 0,
+  },
+  {
+    idx: 10,
+    name: '황병현2',
+    phone: '010-5022-2332',
+    code: '123213',
+    address: '서울 영등포구 선유로 200가길 39',
+    detailAddress: '우형빌딩200호',
+    selected: 0,
+  },
+  {
+    idx: 100,
+    name: '황병현3',
+    phone: '010-5022-2332',
+    code: '123213',
+    address: '서울 영등포구 선유로 200가길 39',
+    detailAddress: '우형빌딩200호',
+    selected: 1,
+  },
+];
+
+export type ShipType = {
+  idx?: number;
+  name: string;
+  phone: string;
+  code: string;
+  address: string;
+  detailAddress: string;
+  selected?: number;
+};
 
 const Shipping: FC = () => {
-  return <div>배송지관리 컴포넌트</div>;
+  // const [user] = useUser();
+  const [shipItems, setShipItems] = useState<ShipType[]>([]);
+  const [selectedShipIdx, setSelectedShipIdx] = useState<number>();
+  const [isModalOpen, setIsModalOpen] = useState<boolean>(false);
+  const [isWriteModal, setIsWriteModal] = useState<boolean>(true);
+  const [modifyItem, setmodifyItem] = useState<ShipType>();
+
+  const fetchShipping = async () => {
+    /**
+     * 주소지 가져오기 api 호출
+     * const response = await getShippingItems();
+     * if (response.statusCode === 200) {
+     *  setShipItems(response.data);
+     * }
+     */
+    setShipItems(mockData);
+    const selectedIdx = mockData.findIndex((item) => item.selected === 1);
+    setSelectedShipIdx(mockData[selectedIdx].idx);
+    alert(`기본 설정 idx는 ${mockData[selectedIdx].idx}`);
+  };
+
+  // useEffect(() => {
+  //   if (user) { fetchShipping();}
+  // }, [user]);
+
+  useEffect(() => {
+    fetchShipping();
+  }, []);
+
+  const handleNewBtnClick = () => {
+    alert('모달 오픈');
+    setIsModalOpen(true);
+  };
+
+  const handleUseThisAddress = () => {
+    alert(`기본배송지가 ${selectedShipIdx}로 설정`);
+    /**
+     * 주소지 선택 api 호출
+     * const response = await putShippingItem(selectedShipIdx);
+     * if (response.statusCode === 200) {
+     *  alert('수정이 완료되었습니다.)
+     * }
+     */
+  };
+
+  const changeSelectedBtn = (shipIdx: number) => {
+    alert(`${shipIdx}선택`);
+    setSelectedShipIdx(shipIdx);
+  };
+
+  const removeShippingItem = (shipIdx: number) => {
+    alert(`${shipIdx}삭제`);
+    /**
+     * 삭제 api 호출
+     * const await deleteShippingItem();
+     * if (response.statusCode === 200) {
+     *  fetchShipping()
+     * }
+     */
+  };
+
+  const handleModalClose = () => {
+    setIsModalOpen(false);
+    setIsWriteModal(true);
+    setmodifyItem(null);
+  };
```

Comment:
오... 이렇게 로직 구분해놓는건 참 좋네요
다만 저라면 setModifyItem 이라고 했을 것 같습니다 :)

```suggestion
  const handleModalClose = () => {
    setIsModalOpen(false);
    setIsWriteModal(true);
    setModifyItem(null);
  };
```

Reply:
크흡 이것도 실수네요. 감사합니다 수정할게요:)